app/jmeter/jira.jmx

```diff
@@ -6848,14 +6878,22 @@ vars.put(&quot;extend_action&quot;, extend_action);
               <boolProp name="TransactionController.parent">true</boolProp>
             </TransactionController>
             <hashTree>
-              <RandomVariableConfig guiclass="TestBeanGUI" testclass="RandomVariableConfig" testname="random page number (1 - 100)" enabled="true">
-                <stringProp name="variableName">page_number</stringProp>
-                <stringProp name="outputFormat"></stringProp>
-                <stringProp name="minimumValue">1</stringProp>
-                <stringProp name="maximumValue">100</stringProp>
-                <stringProp name="randomSeed"></stringProp>
-                <boolProp name="perThread">false</boolProp>
-              </RandomVariableConfig>
+              <BeanShellPreProcessor guiclass="TestBeanGUI" testclass="BeanShellPreProcessor" testname="get project pages count" enabled="true">
+                <boolProp name="resetInterpreter">false</boolProp>
+                <stringProp name="parameters"></stringProp>
+                <stringProp name="filename"></stringProp>
+                <stringProp name="script">import org.apache.commons.io.FileUtils;
+
+int projects = FileUtils.readLines(new File(vars.get(&quot;project_keys_csv&quot;))).size();
+int pages;
+int page_size = 25;
+if (projects % page_size == 0 ){
```

Comment:
you can change long if to ternary operator and code can looks like:
`  int projects = FileUtils.readLines(new File(vars.get("project_keys_csv"))).size();        
   int page_size = 25;
   int pages = (projects % page_size == 0) ? projects / page_size : projects / page_size + 1;
   vars.put("pages", String.valueOf(pages));`


Also you can try to change: 
`int projects = FileUtils.readLines(new File(vars.get("project_keys_csv"))).size();`
to
`long projects = Files.lines(Paths.get("project_keys_csv")).count();`
But then do not forget change `int pages` to `long pages`

Reply:
The ternary operator looks good! Thx for advice )